lib/ansible/modules/windows/win_wait_for_process.ps1

```diff
@@ -0,0 +1,152 @@
+#!powershell
+# This file is part of Ansible
+
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Module Ansible.ModuleUtils.FileUtil
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+
+$process_name_exact = Get-AnsibleParam -obj $params -name "process_name_exact" -type "list" 
+$process_name_pattern = Get-AnsibleParam -obj $params -name "process_name_pattern" -type "str" 
+$process_id = Get-AnsibleParam -obj $params -name "process_id" -type "int" -default 0 #pid is a reserved variable in PowerShell.  use process_id instead.
+$owner = Get-AnsibleParam -obj $params -name "owner" -type "str"
+$sleep = Get-AnsibleParam -obj $params -name "sleep" -type "int" -default 1
+$pre_wait_delay = Get-AnsibleParam -obj $params -name "pre_wait_delay" -type "int" -default 0
+$post_wait_delay = Get-AnsibleParam -obj $params -name "post_wait_delay" -type "int" -default 0
+$process_min_count = Get-AnsibleParam -obj $params -name "process_min_count" -type "int" -default 1
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent"
+$timeout = Get-AnsibleParam -obj $params -name "timeout" -type "int" -default 300
+
+$result = @{
+    changed = $false
+}
+
+# validate the input
+if ($state -eq "absent" -and $sleep -ne 1)
+{
+    Fail-json $result "sleep parameter is of no effect when waiting for a process to stop."
+}
+
+if ($state -eq "absent" -and $process_min_count -ne 1)
+{
+    Fail-json $result "process_min_count parameter is of no effect when waiting for a process to stop."
+}
+
+if (($process_name_exact -or $process_name_pattern) -and $process_id)
+{
+    Fail-json $result "process_id may not be used with process_name_exact or process_name_pattern"
+}
+if ($process_name_exact -and $process_name_pattern)
+{
+    Fail-json $result "process_name_exact and process_name_pattern may not be used at the same time."
+}
+
+if (-not ($process_name_exact -or $process_name_pattern -or $process_id -or $owner))
+{
+    Fail-json $result "at least one of: process_name_exact, process_name_pattern, process_id, or user must be supplied"
```

Comment:
For clarity please change 'user' to 'owner' as that is the name of the module parameters.

Also a really tiny thing... some of the failure messages end in a '.' and others don't.  I don't think we have a particular style but its nice to be consistent.